### ISSUE TITLE
Does mlflow  support DAG workflow？

### ISSUE THREAD

**Issue by frankzqy** [mlflow/mlflow#1563]
MLFlow 怎么支持DAG workflow，if ture，please show an example

**Comment by lorenzwalthert** [comment]
I also wondered and I don't think so. it would be amazing if there would be support for this. The `MLproject` file could host a `dependencies` key like a `MAKE` file.

```yaml
entry_points:
  prepare_clean:
    parameters: 
      y: {type: string, default: "3"}
    command: ...
  clean:
    parameters: 
      x: {type: string, default: "new"}
    command: ...
    dependencies: 
    - prepare_clean

```

Supporting this would be quite an effort though I think. I wonder if there are other ways to combine the strengths of DAG computing with mlflow, notably integrate it in existing workflow managers.

**Comment by frankzqy** [comment]
Thanks. I think  it's important in ML pipeline. I hope MLFLOW could support DAG  in  the next editon ,or  it  have a integrate  solution  wtih existing workflow managers.

**Comment by mparkhe** [comment]
This is a great suggestion. 

For now, you can check out [Multistep Workflow example](https://github.com/mlflow/mlflow/tree/master/examples/multistep_workflow) that uses code and artifacts to define a workflow.

**Comment by lorenzwalthert** [comment]
I believe some soft of DAG processing has been announced in the [Spark AI Summit Keynote on MLflow ](https://youtu.be/QJW_kkRWAUs?t=1788).

**Comment by nehalecky** [comment]
@lorenzwalthert, I was also there and saw that presentation—it was amazing! Is there anyone here that can reference a proposal docs and / or branch that contains that feature development?

**Comment by arita37** [comment]
Very important part, it would be useful

**Comment by pogil** [comment]
Better support for multi-step workflows is still on our long-term roadmap. However, we don't anticipate making significant progress on this until later this year.

**Comment by Samreay** [comment]
+1ing this issue as hard as I can.

To those on the issue waiting in keen anticipation of the DAG workflows teased [here](https://www.youtube.com/watch?v=QJW_kkRWAUs&t=29m0s), what are your current best solutions? Airflow + mlflow?

**Comment by arita37** [comment]
Closer to DAG process is Airflow.

Having the whole pipeline in json,
is th best : flexible and reduce boilerplate code

**Comment by tfurmston** [comment]
Hi, 

I would also be interested in getting an update of multi-step pipelines in mlflow. 

The existing multi-step workflow in the examples has some issues (as noted in the example itself) which are really not ideal. Currently libraries such as DVC seem to handle this case more elegantly. 

It would be great to get an update of the current status of multi-step pipelines in mlflow, along with any roadmap. 

Thanks

**Comment by echarso** [comment]
Yup I also think that this should come fast , perhaps an integration with an existing good solution would be the best here ( airflow , Kubeflow Pipelines or something else)

Thanks

**Comment by nelsoncardenas** [comment]
This would be awesome, or at least to produce something similar to the W&B diagrams

**Comment by BenWilson2** [comment]
This is being addressed in MLflow Pipelines #5456 . Please refer to the pinned issue for further discussion on this topic.
